src/python/gudhi/representations/archipelago.py

```diff
@@ -0,0 +1,119 @@
+        self.archipelago_ = {}
+        self._running_transform_names = ""
+
+    def fit(self, X, y=None):
+        """
+        Calibration step: create and fit `island` vectorizer to each matching diagram element
+
+        Args:
+            X (list or pandas.Series of diagrams): input persistence diagrams to fit vectorizers on.
+            y: possibly labels for each diagram
+
+        Returns:
+            self
+        """
+
+        max_dimension = max(dim for pdiagram in X for (dim, _) in pdiagram)
+        by_dim_list_pdiags = [[
+            np.array([_ for (dim, _) in pdiagram if dim == dimension]) for dimension in range(0, max_dimension + 1)
+        ] for pdiagram in X]
+        for dimension in range(0, max_dimension + 1):
+            this_dim_list_pdiags = [pdiags[dimension] for pdiags in by_dim_list_pdiags]
+            if not len(this_dim_list_pdiags):
+                continue
+            if self.island_dict is not None and dimension in self.island_dict.keys():
+                island = self.island_dict[dimension]
+            elif self.island_dict is not None:
+                continue
+            else:
+                island = copy.deepcopy(self.island)
+            try:
+                island.fit(X=this_dim_list_pdiags, y=y)
+            except ValueError as ve:
+                print(f"[Archipelago] Fit of homology dimension {dimension} returned {ve}. Will ignore this key.")
+                continue
+            print(f"[Archipelago] Fit of homology dimension {dimension} with object {island.__class__} succeeded.")
+            self.archipelago_[dimension] = island
+        return self
+
+    def transform(self, X, y=None):
+        """
+        Apply measure vectorisation on a dictionary of list of measures.
+
+        Args:
+            X (list or pandas.Series of diagrams): input persistence diagrams to vectorize.
+            y: Ignored, present for API consistency by convention.
+
+        Returns:
+            pandas.DataFrame with the same index as X but vectorized columns in order of increased dimension.
+        """
+
+        max_dimension = max(dim for pdiagram in X for (dim, _) in pdiagram)
+        by_dim_list_pdiags = [[
+            np.array([_ for (dim, _) in pdiagram if dim == dimension]) for dimension in range(0, max_dimension + 1)
+        ] for pdiagram in X]
+
+        archipelago_vectorized = []
+        running_transform_names = []
+        for dimension in range(0, max_dimension + 1):
+            if dimension not in self.archipelago_.keys():
+                print(f"[Archipelago] Encounters homology dimension {dimension} that has not been fitted on. Will ignore this key")
+                continue
+            this_dim_list_pdiags = [pdiags[dimension] for pdiags in by_dim_list_pdiags]
+            vectorized_dgms = self.archipelago_[dimension].transform(this_dim_list_pdiags)
+            running_transform_names += [f"{dimension} Center {i + 1}" for i in range(vectorized_dgms.shape[1])]
```

Comment:
Maybe "feature" is more generic than "center" if the island is not necessarily atol. Although the name archipelago may look strange if we don't use atol at all.